packages/ui5-application-writer/src/data/defaults.ts

```diff
@@ -120,22 +121,43 @@ export function mergeUi5(ui5: Partial<UI5>): UI5 {
 }
 
 /**
- * Get the best types version for the given minUI5Version.
- * For the latest versions the LTS S/4 on-premise version (1.102.x) is used, for anything before we match the versions as far back as available.
+ * Get the best types version for the given minUI5Version for https://www.npmjs.com/package/@sapui5/ts-types where specific versions are missing.
  *
- * @param minUI5Version the mininum UI5 version that needs to be supported
+ * @param minUI5Version the minimum UI5 version that needs to be supported
  * @returns semantic version representing the types version.
  */
 export function getTypesVersion(minUI5Version?: string) {
     const version = semVer.coerce(minUI5Version);
-    if (!version) {
+    if (!version || semVer.gte(version, UI5_DEFAULT.TYPES_VERSION_BEST)) {
```

Comment:
This or condition `|| semVer.gte(version, UI5_DEFAULT.TYPES_VERSION_BEST)` was left out on purpose. The reasoning is that it is not possible (with reasonable effort) to ship a type fix for a none LTS version once it is shipped, therefore, the best option for a project that e.g. uses UI5 version `1.103` is to use `1.02.x`, therefore, please remove it again.
Btw, we go a similar approach with our versions for the @sap/ux-specification.